extensions/query-store/src/settings/configureDialog.ts

```diff
@@ -0,0 +1,626 @@
+	private criteriaBasisAvgRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisMaxRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisMinRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisStdDevRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisTotalRadioButton?: azdata.RadioButtonComponent;
+	private executionCountCheckBox?: azdata.CheckBoxComponent;
+	private durationCheckBox?: azdata.CheckBoxComponent;
+	private CPUTimeCheckBox?: azdata.CheckBoxComponent;
+	private logicalReadsCheckBox?: azdata.CheckBoxComponent;
+	private logicalWritesCheckBox?: azdata.CheckBoxComponent;
+	private physicalReadsCheckBox?: azdata.CheckBoxComponent;
+	private CLRTimeCheckBox?: azdata.CheckBoxComponent;
+	private DOPCheckBox?: azdata.CheckBoxComponent;
+	private memoryConsumptionCheckBox?: azdata.CheckBoxComponent;
+	private rowCountCheckBox?: azdata.CheckBoxComponent;
+	private showChartComponent?: azdata.FormComponent<azdata.Component>;
+	private timeIntervalOptionsDropdown?: azdata.DropDownComponent;
+	private customTimeFromTextBox?: azdata.InputBoxComponent;
+	private customTimeToTextBox?: azdata.InputBoxComponent;
+	private localTimeFormatRadioButton?: azdata.RadioButtonComponent;
+	private UTCTimeFormatRadioButton?: azdata.RadioButtonComponent;
+	private timeIntervalComponent?: azdata.FormComponent<azdata.Component>;
+	private aggregationSizeComponent?: azdata.FlexContainer;
+	private timeIntervalOptionsOverallReport?: string[];
+	private timeIntervalOptionsTopResourceReport?: string[];
+	private returnDataAllRadioButton?: azdata.RadioButtonComponent;
+	private returnDataTopRadioButton?: azdata.RadioButtonComponent;
+	private returnDataTopInputBox?: azdata.InputBoxComponent;
+	private returnComponent?: azdata.FormComponent<azdata.Component>;
+	private filtersInputBox?: azdata.InputBoxComponent;
+	private filterComponent?: azdata.FormComponent<azdata.Component>;
+	private initDialogComplete?: Deferred<void>;
+	private initDialogPromise: Promise<void> = new Promise<void>((resolve, reject) => this.initDialogComplete = { resolve, reject });
+
+	constructor() {
+		this.dialog = azdata.window.createModelViewDialog(constants.configure);
+		this.configureTab = azdata.window.createTab(constants.configure);
+		this.dialog.registerCloseValidator(async () => {
+			return this.validate();
+		});
+	}
+
+	async validate(): Promise<boolean> {
+		return true;
+	}
+
+	public async openDialog(): Promise<void> {
+		await this.initializeDialog();
+
+		this.dialog.okButton.label = constants.okButtonText;
+		this.dialog.okButton.enabled = false;
+		this.toDispose.push(this.dialog.okButton.onClick(async () => { }));		// TODO: Add return of settings change and functionality to enable OK Button
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+		this.toDispose.push(this.dialog.cancelButton.onClick(async () => await this.cancel()));
+
+		let applyButton = azdata.window.createButton(constants.applyButtonText);
```

Comment:
what's the difference between the apply and ok button?

Reply:
In SSMS, Apply button updates the UI with the changes made in the dialog (in the background), but keeps the config dialog open. Ok = Apply functionality + Close the dialog.
If it isn't necessary, I can remove it. I just wanted a parity between SSMS and ADS, hence this button :).

Reply:
ah ok got it. A little confusing to a new person to me, but makes sense to have parity with SSMS.

Reply:
On a different thought, I feel "Apply" along with "Ok" button will be confusing for ADS users. Even thought Ok + Cancel + Apply is a frequent siting in SSMS, but there is none in ADS. And Apply in ADS = Ok button. So for less confusion, I will remove the Apply button. And we can reintroduce it if it makes sense later.

Reply:
Removed Apply button, since it doesn't exist in ADS (and can confuse ADS customers).